code/game/objects/items/implants/implant_mindshield.dm

```diff
@@ -26,6 +26,16 @@
 		if(target.mind.has_antag_datum(/datum/antagonist/brainwashed))
 			target.mind.remove_antag_datum(/datum/antagonist/brainwashed)
 
+		if(is_hivemember(target))
+			var/warning = ""
+			for(var/datum/antagonist/hivemind/hive in GLOB.antagonists)
+				if(hive.hivemembers.Find(target))
+					var/datum/mind/M = hive.owner
+					if(M)
+						warning += "[M.current.real_name]. "
```

Comment:
Check if the mind still has current at this point.

Reply:
Done